### PR TITLE
[JENKINS-55664] Include developer documentation

### PR DESCRIPTION
**Pull request by varyvol**
[JENKINS-55664](https://issues.jenkins-ci.org/browse/JENKINS-55664)

Include documentation that can be useful for developers to contribute to the project.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  * Any IDE which supports importing Maven projects.
+  * Docker.
+  * Make.
+
+## Building and Debugging
+
+The build flow for Jenkinsfile Runner is built around Maven and Docker.
+
+Run `mvn clean package` for the project to be just built and `mvn clean package -Denvironment=test` for the project
+to be built and the integration tests, based on the [Jenkinsfile Runner Test Framework](https://github.com/jenkinsci/jenkinsfile-runner-test-framework), to be launched.
```
Comment (fcojfernandez) [inline CONTRIBUTING.md]: The commas and pauses make the reading difficult. You might loose the idea of being launching the tests. What about something like this?
```suggestion
to be built and the integration tests to be launched, which are based on the [Jenkinsfile Runner Test Framework](https://github.com/jenkinsci/jenkinsfile-runner-test-framework).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+3. Select `jenkinsci/jenkinsfile-runner` as _base fork_ and `master` as `base`, then click _Create Pull Request_
+4. Fill in the Pull Request description according to the changes.
+5. Click _Create Pull Request_
+6. Wait for CI results/reviews, process the feedback.
+
+It is recommended that new features/changes include testing proving the changes are working correctly.
+
+## Continuous Integration
+
+Jenkinsfile Runner uses [ci.jenkins.io](http://ci.jenkins.io) as Continuous Integration server and uses Jenkins Pipeine to run builds.
```
Comment (fcojfernandez) [inline CONTRIBUTING.md]:
```suggestion
Jenkinsfile Runner uses [ci.jenkins.io](http://ci.jenkins.io) as Continuous Integration server and uses Jenkins Pipeline to run builds.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+The code for the build flow is stored in the [Jenkinsfile](Jenkinsfile) in the repository root.
```
Comment (fcojfernandez) [inline CONTRIBUTING.md:60]: What do you mean here?

File: DEVELOPER.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+Contains the configuration for the "real" pipeline job to be configured and run on the ephemeral Jenkins instance. It basically:
+  * Creates a Pipeline job configured to optimize performance.
+  * Sets the Jenkinsfile to be executed.
+  * Sets the specified parameters.
+  * (Un)Sets the script security plugin security layer.
+  * Schedules a build and wait for the finalisation, redirecting the logs to the system output.
+
+## `payload-dependencies` module
+
+The `payload` module needs to depend on the presence of many other plugins,
```
Comment (fcojfernandez) [inline DEVELOPER.md]: It sounds weird to me and maybe I'm misunderstanding you. What about this?
```suggestion
The `payload` module depends on the presence of many other plugins,
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (varyvol) [reply CONTRIBUTING.md:60]: That the build flow/process is defined in the Jenkinsfile and thus can be changed if required through a PR, as any other change.

Comment (fcojfernandez) [reply CONTRIBUTING.md:60]: Clarified!

**Review by fcojfernandez — APPROVED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+This page provides information about contributing code to the Jenkinsfile Runner codebase.
+
+## Getting started
+
+1. Fork the repository on GitHub
+2. Clone the forked repository to your machine
+3. Install the development tools. In order to develop Jenkinsfile Runner, you need the following tools:
+  * Java Development Kit (JDK) 8.
+     - We usually use [OpenJDK](http://openjdk.java.net/) but you can use other JDKs as well.
+  * Maven 3.5.3 or above.
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md]: Better to recommend 2.5.4, but not a big difference

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  * Any IDE which supports importing Maven projects.
+  * Docker.
+  * Make.
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md:15]: I have some difficulties with the current tests & Windows, but I believe I should submit a bug first :D

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+3. Select `jenkinsci/jenkinsfile-runner` as _base fork_ and `master` as `base`, then click _Create Pull Request_
+4. Fill in the Pull Request description according to the changes.
+5. Click _Create Pull Request_
+6. Wait for CI results/reviews, process the feedback.
+
+It is recommended that new features/changes include testing proving the changes are working correctly.
+
+## Continuous Integration
+
+Jenkinsfile Runner uses [ci.jenkins.io](http://ci.jenkins.io) as Continuous Integration server and uses Jenkins Pipeline to run builds.
```
Comment (oleg-nenashev) [inline CONTRIBUTING.md:58]:
```suggestion
Jenkinsfile Runner uses [ci.jenkins.io](http://ci.jenkins.io) as Continuous Integration server and uses Jenkins Pipeline to run builds.
See https://ci.jenkins.io/blue/organizations/jenkins/Tools%2Fjenkinsfile-runner/ .
```

**Review by oleg-nenashev — APPROVED** [review]
looks great, thanks!
Just few minor comments

File: CONTRIBUTING.md (hunk 5 above)
Comment (varyvol) [reply CONTRIBUTING.md]: Isn't that one too old? I don't think it hurts to recommend a newer one.

Comment (oleg-nenashev) [reply CONTRIBUTING.md]: Sorry, 3.5.4

```suggestion
  * Maven 3.5.4 or above.
```

**Pull request merged**